app/src/main/java/com/example/drawingactivity/DialogUtils.kt

```diff
@@ -0,0 +1,151 @@
+                layerNameInput.setText(viewModel.userNewLayerName)
+                errorView.isVisible =
+                    layerNameInput.text.toString().length > DrawViewModel.CHAR_LIMIT
+            }
+            layerNameInput.addTextChangedListener {
+                viewModel.userNewLayerName = it.toString().trim()
+                errorView.isVisible = it.toString().trim().length > DrawViewModel.CHAR_LIMIT
+            }
+            setPositiveButton(context.getString(R.string.addLayerPositiveButton)) { _, _ ->
+                var layerName: String = layerNameInput.text.toString().trim()
+                when (layerName) {
+                    context.getString(R.string.layerHint, viewModel.hintIndex) -> {
+                        viewModel.addLayer(layerName)
+                    }
+                    DrawViewModel.NAME_EMPTY -> {
+                        layerName = context.getString(R.string.layerHint, viewModel.hintIndex)
+                        viewModel.addLayer(layerName)
+                    }
+                    else -> {
+                        viewModel.addLayer(layerName)
+                    }
+                }
+                viewModel.dialogCompleted()
+            }
+            setNegativeButton(context.getString(R.string.addLayerNegativeButton)) { _, _ ->
+                viewModel.dialogCompleted()
+            }
+            setOnCancelListener {
+                viewModel.dialogCompleted()
+            }
+        }
+        dialog.show()
+    }
+
+    fun editLayerDialog(index: Int) {
+        viewModel.isEditDialogOpen = true
+        val inflater = LayoutInflater.from(context)
+        val view = inflater.inflate(R.layout.add_layer_dialog, null)
+        val dialog = AlertDialog.Builder(context).apply {
+            setTitle(context.getString(R.string.editLayerDialogTitle))
+            setView(view)
+            val errorView = view.findViewById<TextView>(R.id.errorText)
+            val editView = view.findViewById<EditText>(R.id.layerNameField)
+            while (!viewModel.layerHintFound) {
+                val hint = context.getString(R.string.layerHint, viewModel.hintIndex)
+                viewModel.createLayerHint(hint)
+            }
+            editView.hint = context.getString(R.string.layerHint, viewModel.hintIndex)
+            if (viewModel.isEditDialogOpen && viewModel.userNewLayerName != null) {
+                editView.setText(viewModel.userNewLayerName)
+                errorView.isVisible = editView.text.toString().length > DrawViewModel.CHAR_LIMIT
+            }
+            editView.addTextChangedListener {
+                viewModel.userNewLayerName = it.toString().trim()
+                errorView.isVisible =
+                    editView.text.toString().trim().length > DrawViewModel.CHAR_LIMIT
+            }
+            viewModel.editLayerIndex = index
+            setPositiveButton(context.getString(R.string.addLayerPositiveButton)) { _, _ ->
```

Comment:
Ideally you really just plug this information into the addLayerDialog function by using lambdas and just have a call to it in the bottomSheet when they click edit.